lib/convertkit/connection.rb

```diff
@@ -14,14 +15,12 @@ def initialize(url, options = {})
       end
     end
 
-    def get(path, params = {})
-      response = @connection.get(path, process_request(params))
-      process_response(response)
-    end
-
-    def post(path, params = {})
-      response = @connection.post(path, process_request(params))
-      process_response(response)
+    # Defined wrapper methods for Faraday's HTTP methods
+    HTTP_METHODS.each do |method|
+      define_method(method) do |path, params = {}|
+        response = @connection.public_send(method, path, process_request(params))
+        process_response(response)
```

Comment:
since this is repeated, can it live in an helper?

Reply:
The logic is slightly different. In the connection class we process both the request and response while in the client class we only care about the response. Even thought we could abstract this into a helper method with callbacks, I don't think it's worth the effort. It's only going to be used in these 2 classes anyways.

Reply:
Agreed on keeping it inline for now. It'll become difficult to find otherwise where the methods were defined. Already dynamically defining methods makes it a little difficult to use the IDE and further burying this into another helper will make it more difficult.